text-editor/text-editor-002-en.md

```diff
@@ -5,68 +5,70 @@
+business, they also post some `news`. News are small texts, which may contain
+tweets or links to other articles.
+
+Editors and web managers needed a tool to be able to create and edit these kinds
+of contents. These contents are saved in our database as an HTML string.
+
+There alreay is many text edition tools online, but we wanted a tool which
+looks close to the modern text editor of [medium.com](https://medium.com/).
+Some open source libraries are offering such solution, but most of the time,
+when specific needs arise, these libraries no longer fulfill all your
+requirements. And here you are on the highway of awfull tweaks... Instead of
+spending time to develop additional features on top of an existing library (with
+all drawbacks implied), we decided to implement our own tool. That way, we could
+make it behave exactly the way we want.
+
+Keep in mind that our text editor development took place whereas the frontend
+website used to publish informations was close to be put online.
+
+Technically speaking, articles are fetched from an HTTP REST API (our backend).
+Some of them were coming from a legacy application, and may contain some broken
+HTML semantic. To address this problem, we decided to develop a parser for the
+received HTML markup to be cleaned. This parser creates React components for,
+and only for HTML tags that we allow. Doing so, only tags that we're sure to
+support are rendered, which also simplifies design and stylization.
+
+The same parsing logic had to be kept for our text editor as we wanted it to be
+WYSIWYG, i.e. display the edited content like it would be displayed in the
+public website, with the ability to directly edit it.
 
 ## The specs
 
-The text editor sould provide the basic fatures of text edition, ie be able to :
+The text editor sould provide the basic fatures of text edition :
 - apply some style to a portion of text (bold, italic, underlined)
 - create an HTTP link
 - create some paragraphs
 - create titles
+- create quotes
 
-There are also some media features :
-- be able to insert an image (upload it or pick it from the image collection
-of the company)
-- be able to insert a video (from the video collection of the company)
-- be able to insert a tweet (by specifying the tweet URL)
-- be able to insert a Youtube video (by specifying the video URL)
+Our client also needed to insert some more specific medias :
+- regular images (uploaded or picked from their image collection)
+- videos (also picked from their video collection)
+- tweets (by pasting the tweet URL)
+- Youtube videos (by pasting the video URL)
 
-Such features may seems very common, but they are however tighten to our
-current application as images and videos are dealing with our image and videos
-collection.
+Although such features may seems very common, complexity shows up because they
+are tighten to our current application (images and videos collections are parts
+of the application state).
```

Comment:
Are you refering to the redux state ? I don't think this is a difficult point. If images were coming from a public repo, we also had to store the fetched collection representation in the app's state.
What I wanted to say in the first place, is that the image picking feature was tighten to our image collection from the DB, so it was **our** collection (ie we have to deal with our data model and make calls to our backend to fetch images). It's not only a file browser as we can expect from regular text editon libraries. I wanted to put in the light the fact that we had to interface with something that is specific to our current system.

Reply:
Thanks for clarification, I will reformulate that :)